### PR TITLE
Feature/user info modal

### PR DESCRIPTION
**Pull request by ReginaVarybrus**
Added UI user's info modal window

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,10 @@
   "dependencies": {
     "@emotion/react": "^11.11.1",
     "@emotion/styled": "^11.11.0",
+    "@esbuild/win32-x64": "^0.20.2",
```
Comment (DimaShubin) [inline package.json]: Here

File: package.json [hunk 2]
```diff
@@ -6,8 +6,10 @@
...
     "@mui/material": "^5.14.15",
     "@reduxjs/toolkit": "^2.2.1",
+    "@stomp/stompjs": "^7.0.0",
```
Comment (DimaShubin) [inline package.json]: Here

File: package.json [hunk 3]
```diff
@@ -34,6 +36,7 @@
     "vite": "^5.0.12",
     "vite-tsconfig-paths": "^4.3.1",
     "web-vitals": "^2.1.4",
+    "websocket": "^1.0.34",
```
Comment (DimaShubin) [inline package.json]: Here

File: src/components/SearchInput/SearchInput.jsx [hunk 4]
```diff
@@ -50,6 +50,11 @@ const SearchInput = ({ onCountryRoomDataReceived }) => {
 
   const handleClick = () => setShowItem(!showItem);
 
+  const onDataReceived = data => {
+    console.log('recieved data:', data);
```
Comment (DimaShubin) [inline src/components/SearchInput/SearchInput.jsx]: ???

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
+// import { useState } from 'react';
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]: ??

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  ModalAvatar,
+  AboutUser,
+  InfoIcon,
+  ButtonBlock,
+} from './UserInfoModalStyled';
+
+const UserInfoModal = ({
+  open,
+  handleClose,
+  avatar,
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx:21]:
```suggestion
  avatar = avatarImage,
```

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      },
+    }}
+  >
+    <Fade in={open}>
+      <InfoModal>
+        <ButtonClose onClick={handleClose} />
+        <UserContactInfo>
+          <ModalAvatar src={avatar || avatarImage} alt="avatar" />
+          <div>
+            <h5>{userName || 'User name'}</h5>
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]: Use default value for props as I proposed to "avatar" above.

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            <p>{userEmail || 'email@gmail.com'}</p>
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]: Default value

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    closeAfterTransition
+    slots={{ backdrop: Backdrop }}
+    slotProps={{
+      backdrop: {
+        timeout: 500,
+      },
+    }}
+  >
+    <Fade in={open}>
+      <InfoModal>
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]: InfoModalStyled

File: package.json (hunk 1 above)
Comment (ReginaVarybrus) [reply package.json]: Removed

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        <ButtonClose onClick={handleClose} />
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]: This must be button.

File: package.json (hunk 2 above)
Comment (ReginaVarybrus) [reply package.json]: Removed

File: package.json (hunk 3 above)
Comment (ReginaVarybrus) [reply package.json]: Removed

File: src/components/UserInfoModal/UserInfoModalStyled.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
+import styled from 'styled-components';
+import Box from '@mui/material/Box';
+import { IoCloseOutline } from 'react-icons/io5';
+import { LuInfo } from 'react-icons/lu';
+import { device } from '@/constants/mediaQueries';
+
+export const InfoModal = styled(Box)`
+  box-sizing: border-box;
+  position: absolute;
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModalStyled.js]: This component could be centered without position: absolute.

File: src/components/UserInfoModal/UserInfoModalStyled.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+`;
+
+export const ButtonClose = styled(IoCloseOutline)`
+  position: absolute;
+  top: 20px;
+  right: 20px;
+  width: 24px;
+  height: 24px;
+  &:hover {
+    cursor: pointer;
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModalStyled.js]: Use button

File: src/components/UserInfoModal/UserInfoModal.jsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        <UserContactInfo>
+          <ModalAvatar src={avatar || avatarImage} alt="avatar" />
```
Comment (DimaShubin) [inline src/components/UserInfoModal/UserInfoModal.jsx]:
```suggestion
          <ModalAvatar src={avatar} alt="avatar" />
```

File: src/components/SearchInput/SearchInput.jsx (hunk 4 above)
Comment (ReginaVarybrus) [reply src/components/SearchInput/SearchInput.jsx]: Omg, removed

File: src/components/UserInfoModal/UserInfoModal.jsx (hunk 5 above)
Comment (ReginaVarybrus) [reply src/components/UserInfoModal/UserInfoModal.jsx]: Removed

File: src/components/UserInfoModal/UserInfoModal.jsx (hunk 7 above)
Comment (ReginaVarybrus) [reply src/components/UserInfoModal/UserInfoModal.jsx]: Ok

File: src/components/UserInfoModal/UserInfoModal.jsx (hunk 6 above)
Comment (ReginaVarybrus) [reply src/components/UserInfoModal/UserInfoModal.jsx:21]: Done

File: src/components/MessageItem/MessageItem.jsx [hunk 14]
```diff
@@ -59,19 +58,16 @@ const MessageItem = ({
           <ContentMessage>{content || `message`}</ContentMessage>
           <Time>{time || 'time'}</Time>
         </MessageContent>
-
+      )}
+      {type === 'JOIN' && <ContentJoin>{content || `message`}</ContentJoin>}
```
Comment (DimaShubin) [inline src/components/MessageItem/MessageItem.jsx]: Would be better to create a constant to store these types.

File: src/components/MessageItem/MessageItem.jsx [hunk 15]
```diff
@@ -6,13 +6,12 @@ import ULRs from '@/redux-store/constants';
...
 import PropTypes from 'prop-types';
 import UserInfoModal from '../UserInfoModal/UserInfoModal';
-import UserAvatar from '../UserAvatar/UserAvatar';
 import { timeStampConverter } from '../utils/timeUtil.js';
 
 import {
   MessageItemStyled,
   MessageContent,
-  Avatar,
+  LetterAvatar,
```
Comment (DimaShubin) [inline src/components/MessageItem/MessageItem.jsx]: Regular component or styled component?

Comment (ReginaVarybrus) [reply src/components/MessageItem/MessageItem.jsx]: Styled component

**Review by DimaShubin — APPROVED** [review]

**Pull request merged**